src/Lecture4.hs

```diff
@@ -264,7 +334,10 @@ Use functions 'readFile' and 'putStrLn' here.
 -}
 
 printProductStats :: FilePath -> IO ()
-printProductStats = error "TODO"
+printProductStats filePath = do
+      fileData <- readFile filePath
+      let result = calculateStats fileData
+      putStrLn result
```

Comment:
Perfect 🏆 
Absolutely glorious implementation of "Function Core, Imperative Shell" pattern ❇️